framework/include/utils/MooseMeshElementConversionUtils.h

```diff
@@ -0,0 +1,206 @@
+ * @param min_id_index The index of the node with the minimum id
+ * @return a vector of the three neighboring nodes
+ */
+std::vector<unsigned int> neighborNodeIndicesHEX8(unsigned int min_id_index);
+
+/**
+ * For a vector of rotated nodes that can form a HEX8 element, create a vector of four-node sets
+ * that can form TET4 elements to replace the original HEX8 element. All the QUAD4 faces of the
+ * HEX8 element will be split by the diagonal line that involves the node with the minimum id of
+ * that face.
+ * @param hex_nodes A vector of pointers to the nodes that can form a HEX8 element
+ * @param rotated_tet_face_indices A vector of vectors of the original face indices of the HEX8
+ * element corresponding to the faces of the newly created TET4 elements
+ * @param tet_nodes_list a vector of vectors of pointers to the nodes that can form TET4 elements
+ */
+void hexNodesToTetNodesDeterminer(std::vector<const Node *> & hex_nodes,
+                                  std::vector<std::vector<unsigned int>> & rotated_tet_face_indices,
+                                  std::vector<std::vector<const Node *>> & tet_nodes_list);
+
+/**
+ * For a HEX8 element, determine the direction of the diagonal line of each face that involves the
+ * node with the minimum id of that face.
+ * @param hex_nodes A vector of pointers to the nodes that can form a HEX8 element
+ * @return a vector of boolean values indicating the direction of the diagonal line of each face
+ */
+std::vector<bool> quadFaceDiagonalDirectionsHex(const std::vector<const Node *> & hex_nodes);
+
+/**
+ * For a QUAD4 element, determine the direction of the diagonal line that involves the node with
+ * the minimum id of that element.
+ * @param quad_nodes A vector of pointers to the nodes that can form a QUAD4 element
+ * @return a boolean value indicating the direction of the diagonal line
+ */
+bool quadFaceDiagonalDirection(const std::vector<const Node *> & quad_nodes);
+
+/**
+ * Creates sets of four nodes indices that can form TET4 elements to replace the original HEX8
+ * element.
+ * @param diagonal_directions A vector of boolean values indicating the direction of the diagonal
+ * line of each face; true means the diagonal line is connecting node 0 and node 2, false means the
+ * diagonal line is connecting node 1 and node 3 of that quad face
+ * @param tet_face_indices A vector of vectors of the original face indices of the HEX8 element
+ * corresponding to the faces of the newly created TET4 elements
+ * @return a vector of vectors of node indices that can form TET4 elements
+ */
+std::vector<std::vector<unsigned int>>
+tetNodesForHex(const std::vector<bool> diagonal_directions,
+               std::vector<std::vector<unsigned int>> & tet_face_indices);
+
+/**
+ * Rotate a PRISM6 element nodes to ensure that the node with the minimum id is the first node.
+ * @param min_id_index The index of the node with the minimum id
```

Comment:
```suggestion
 * @param min_id_index The index of the node, within the prism nodes, with the minimum id
```

Reply:
trying to convey that it s not the node id but it s the index of the node in the element _nodes vector